main.go

```diff
@@ -101,15 +106,96 @@ func validateMongoURL(mongourl string) error {
+				// An error occurred on the server side.
+				fmt.Println(secretsmanager.ErrCodeInternalServiceError, aerr.Error())
+
+			case secretsmanager.ErrCodeInvalidParameterException:
+				// You provided an invalid value for a parameter.
+				fmt.Println(secretsmanager.ErrCodeInvalidParameterException, aerr.Error())
+
+			case secretsmanager.ErrCodeInvalidRequestException:
+				// You provided a parameter value that is not valid for the current state of the resource.
+				fmt.Println(secretsmanager.ErrCodeInvalidRequestException, aerr.Error())
+
+			case secretsmanager.ErrCodeResourceNotFoundException:
+				// We can't find the resource that you asked for.
+				fmt.Println(secretsmanager.ErrCodeResourceNotFoundException, aerr.Error())
+			}
+		} else {
+			// Print the error, cast err to awserr.Error to get the Code and
+			// Message from an error.
+			fmt.Println(err.Error())
+		}
+		return ""
+	}
+
+	// Decrypts secret using the associated KMS CMK.
+	// Depending on whether the secret is a string or binary, one of these fields will be populated.
+	var secretString, decodedBinarySecret string
+	if result.SecretString != nil {
+		secretString = *result.SecretString
+		return secretString
+	} else {
+		decodedBinarySecretBytes := make([]byte, base64.StdEncoding.DecodedLen(len(result.SecretBinary)))
+		len, err := base64.StdEncoding.Decode(decodedBinarySecretBytes, result.SecretBinary)
+		if err != nil {
+			fmt.Println("Base64 Decode Error:", err)
+			return ""
+		}
+		decodedBinarySecret = string(decodedBinarySecretBytes[:len])
+		return decodedBinarySecret
+	}
+}
+
 func main() {
 	flags := flag.NewFlagSet("whackanop", flag.ExitOnError)
-	mongourl := flags.String("mongourl", "mongodb://localhost?connect=direct",
+	mongourl := flags.String("mongourl", "mongodb://localhost:27017/?connect=direct",
 		"mongo url to connect to. Must specify connect=direct to guarantee admin commands are run on the specified server.")
 	interval := flags.Int("interval", 1, "how often, in seconds, to poll mongo for operations")
 	querystr := flags.String("query", `{"op": "query", "secs_running": {"$gt": 60}}`, "query sent to db.currentOp()")
 	debug := flags.Bool("debug", true, "in debug mode, operations that match the query are logged instead of killed")
 	version := flags.Bool("version", false, "print the version and exit")
 	verbose := flags.Bool("verbose", false, "more verbose logging")
+
+	secretManager := flags.Bool("secret", false, "Enable SecretManager")
+	if *secretManager {
+		secretName := flags.String("secretname", "whackanop",
+			"SecretManager Name to connect")
+		region := flags.String("region", "ap-southeast-1",
+			"SecretManager Name to connect")
+		secretPath := flags.String("secretpath", "local",
+			"SecretManager Path to connect")
+		var secretString = getSecret(*secretName, *region)
+		secretBytes := []byte(secretString)
+		var raw map[string]interface{}
+		if err := json.Unmarshal(secretBytes, &raw); err != nil {
+			panic(err)
```

Comment:
I think the convention seems to be to use `log.Fatal(err)` looking at code in other places.